trade-finance-manager-ui/templates/utilisation-reports/edit-payment.njk

```diff
@@ -0,0 +1,122 @@
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% import "./_macros/fee-record-details-table.njk" as feeRecordDetailsTable %}
+{% extends "index.njk" %}
+{% set backLink = {"href": '/utilisation-reports/' + reportId} %}
+{% block pageTitle %}Edit payment{% endblock %}
+{% block content %}
+  {% if errors.errorSummary | length %}
+    {{ govukErrorSummary({
+        titleText: "There is a problem",
+        errorList: errors.errorSummary,
+        attributes: {
+        'data-cy': 'error-summary'
+        }
+    }) }}
+  {% endif %}
+  <div class="govuk-grid-row govuk-!-padding-top-7">
+    <div class="govuk-grid-column-one-half">
+      <span class="govuk-caption-l" data-cy="edit-payment-heading-caption">{{ bank.name }},
+      {{ formattedReportPeriod }}</span>
+      <h1 class="govuk-heading-l" data-cy="edit-payment-heading">Edit payment</h1>
+    </div>
+  </div>
+  {{ feeRecordDetailsTable.render({
+    captionText: "Added reported fees details",
+    feeRecords: feeRecords,
+    totalReportedPayments: totalReportedPayments,
+    enableSelectingFeeRecords: true
+  }) }}
+  <h2 class="govuk-heading-m">
+    Payment details
+  </h2>
+  <form method="post">
+    <input type="hidden" name="_csrf" value="{{ csrfToken }}"/> {{ govukInput({
+        label: {
+          text: "Amount received",
+          classes: "govuk-label--s"
+        },
+        classes: "govuk-input--width-20",
+        hint: {
+          text: "The currency cannot be changed"
+        },
+        prefix: {
+          text: paymentCurrency
+        },
+        id: "paymentAmount",
+        name: "paymentAmount",
+        value: formValues.paymentAmount,
+        errorMessage: errors.paymentAmountErrorMessage and {
+          text: errors.paymentAmountErrorMessage
+        }
+    }) }}
+    {{ govukDateInput({
+      id: "paymentDate",
+      namePrefix: "paymentDate",
+      fieldset: {
+        legend: {
+          text: "Date payment received",
+          classes: "govuk-fieldset__legend--s"
+        }
+      },
+      hint: {
+        text: "For example, 27 03 2023"
+      },
+      errorMessage: errors.paymentDateError and {
+        text: errors.paymentDateError.message
+      },
+      items: [
+        {
+          classes: (errors.paymentDateError.dayError and "govuk-input--width-2 govuk-input--error") or "govuk-input--width-2",
+          name: "day",
+          value: formValues.paymentDate.day
+        },
+        {
+          classes: (errors.paymentDateError.monthError and "govuk-input--width-2 govuk-input--error") or "govuk-input--width-2",
+          name: "month",
+          value: formValues.paymentDate.month
+        },
+        {
+          classes: (errors.paymentDateError.yearError and "govuk-input--width-4 govuk-input--error") or "govuk-input--width-4",
+          name: "year",
+          value: formValues.paymentDate.year
+        }
+      ]
+    }) }}
```

Comment:
I considered putting this into a shared macro, but decided against it as there are very slight differences which make the shared macro harder to read and use for future cases. As an example of where I did make the shared macro, I created  the fee record table one because there was only one table header which was different, so the single optional `override` parameter felt like it was worth it